android/src/Loader.java

```diff
@@ -29,12 +29,39 @@
   private static final String TAG = "XCSoar";
 
   public static boolean loaded = false;
+  public static boolean isLegacy = false;
   public static String error;
 
   static {
     try {
-      System.loadLibrary("xcsoar");
-      loaded = true;
+      /* Depending on the Android version we load the fully functional library
+       * or the functional slightly restricted version which is compatible down to
+       * API level 12 (Honeycomb), and implements some work-arounds for missing functions
+       * in the Bionic C library
+       */
+
+      if (android.os.Build.VERSION.SDK_INT >= 21) {
+        System.loadLibrary("xcsoar");
+        Log.i(TAG, "Loaded native library xcsoar");
+        loaded = true;
+      }
+    } catch (UnsatisfiedLinkError e) {
+      ;
```

Comment:
Don't just ignore the exception!

Reply:
Will add an appropriate warning log message.